azure-pipelines.yaml

```diff
@@ -0,0 +1,201 @@
+    chmod 777 test-output
+  displayName: create test output directory
+  # build test image
+- task: DockerCompose@0
+  inputs:
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+    dockerComposeCommand: -f docker-compose.test.yaml build --no-cache $(imageName)
+  displayName: build tests image
+  # run tests
+- task: DockerCompose@0
+  inputs:
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+    dockerComposeCommand: -f docker-compose.test.yaml up --force-recreate  $(imageName)
+  displayName: run tests
+  # build base image
+- task: DockerCompose@0
+  inputs:
+    action: Build services
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+    additionalImageTags: $(containerTag)
+  displayName: build base image
+  # push container to registry
+- task: DockerCompose@0
+  inputs:
+    action: Push services
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    qualifyImageNames: true
+    additionalImageTags: $(containerTag)
+  displayName: Push container to registry
+  # publish test results
+- task: PublishTestResults@2
+  displayName: Publish test results
+  inputs:
+    testResultsFormat: JUnit
+    testResultsFiles: test-output/junit.xml
+    failTaskOnFailedTests: true
+    testRunTitle: Unit Test
+  # deploy via helm
+- task: HelmDeploy@0
+  inputs:
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureResourceGroup: $(azureResourceGroup)
+    kubernetesCluster: $(kubernetesCluster)
+    namespace: $(imageName)-$(safeContainerTag)
+    command: upgrade
+    chartType: filepath
+    chartPath: helm
+    releaseName: $(imageName)-$(containerTag)
+    overrideValues: 'container.messageQueueHost=$(messageQueueHost),container.messageQueuePort=$(messageQueuePort),container.messageQueueTransport=$(messageQueueTransport),container.scheduleQueueAddress=$(scheduleQueueAddress),container.scheduleQueueUser=$(scheduleQueueUser),container.scheduleQueuePassword=$(scheduleQueuePassword),container.paymentQueueAddress=$(paymentQueueAddress),container.paymentQueueUser=$(paymentQueueUser),container.paymentQueuePassword=$(paymentQueuePassword),postgresExternalName=$(postgresExternalName),postgresPassword=$(postgresPassword)'

```

Comment:
The docs for `overrideValues` indicate that we could "supply multiple values" rather than compressing it all onto one line, to improve readability:
```
 overrideValues:
  - container.messageQueueHost=$(messageQueueHost)
  - container.messageQueuePort=$(messageQueuePort)
  - container.messageQueueTransport=$(messageQueueTransport)
  - container.scheduleQueueAddress=$(scheduleQueueAddress)
  - container.scheduleQueueUser=$(scheduleQueueUser)
  - container.scheduleQueuePassword=$(scheduleQueuePassword)
  - container.paymentQueueAddress=$(paymentQueueAddress)
  - container.paymentQueueUser=$(paymentQueueUser)
  - container.paymentQueuePassword=$(paymentQueuePassword)
  - postgresExternalName=$(postgresExternalName)
  - postgresPassword=$(postgresPassword)
```

Reply:
Which docs state you can do this?  There's no reference I can find in the Microsoft docs? https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/helm-deploy?view=azure-devops

Reply:
It's on that page in the parameter lists for install and upgrade commands. The description for overrideValues says "Set values on the command line. You can specify multiple values, or separate values with commas."